pkg/operator/controller/ingress/status_test.go

```diff
@@ -800,10 +877,389 @@ func Test_computeLoadBalancerProgressingStatus(t *testing.T) {
+			ic: loadBalancerIngressControllerWithAWSSubnets(
+				operatorv1.AWSClassicLoadBalancer,
+				&operatorv1.AWSSubnets{
+					IDs:   []operatorv1.AWSSubnetID{"subnet-12345"},
+					Names: []operatorv1.AWSSubnetName{"name-12345"},
+				},
+				&operatorv1.AWSSubnets{
+					IDs:   []operatorv1.AWSSubnetID{"subnet-12345"},
+					Names: []operatorv1.AWSSubnetName{"name-12345"},
+				},
+			),
+			service:           &corev1.Service{},
+			awsSubnetsEnabled: true,
+			platformStatus:    awsPlatformStatus,
+			expectStatus:      operatorv1.ConditionFalse,
+		},
+		{
+			name: "CLB LoadBalancerService, AWS Subnets spec and status are NOT equal",
+			ic: loadBalancerIngressControllerWithAWSSubnets(
+				operatorv1.AWSClassicLoadBalancer,
+				&operatorv1.AWSSubnets{
+					IDs:   []operatorv1.AWSSubnetID{"subnet-12345"},
+					Names: []operatorv1.AWSSubnetName{"name-12345"},
+				},
+				&operatorv1.AWSSubnets{
+					IDs:   []operatorv1.AWSSubnetID{"subnet-67890"},
+					Names: []operatorv1.AWSSubnetName{"name-67890"},
+				},
+			),
+			service:           &corev1.Service{},
+			awsSubnetsEnabled: true,
+			platformStatus:    awsPlatformStatus,
+			expectStatus:      operatorv1.ConditionTrue,
+		},
+		{
+			name: "CLB LoadBalancerService, AWS Subnets spec and status are equal with different order",
+			ic: loadBalancerIngressControllerWithAWSSubnets(
+				operatorv1.AWSClassicLoadBalancer,
+				&operatorv1.AWSSubnets{
+					IDs:   []operatorv1.AWSSubnetID{"subnet-12345", "subnet-67890"},
+					Names: []operatorv1.AWSSubnetName{"name-12345", "name-67890"},
+				},
+				&operatorv1.AWSSubnets{
+					IDs:   []operatorv1.AWSSubnetID{"subnet-67890", "subnet-12345"},
+					Names: []operatorv1.AWSSubnetName{"name-67890", "name-12345"},
+				},
+			),
+			service:           &corev1.Service{},
+			awsSubnetsEnabled: true,
+			platformStatus:    awsPlatformStatus,
+			expectStatus:      operatorv1.ConditionFalse,
+		},
+		{
+			name: "CLB LoadBalancerService, AWS Subnets spec and status have extra items",
+			ic: loadBalancerIngressControllerWithAWSSubnets(
+				operatorv1.AWSClassicLoadBalancer,
+				&operatorv1.AWSSubnets{
+					IDs:   []operatorv1.AWSSubnetID{"subnet-67890", "subnet-12345", "subnet-54321"},
+					Names: []operatorv1.AWSSubnetName{"name-12345", "name-67890"},
+				},
+				&operatorv1.AWSSubnets{
+					IDs:   []operatorv1.AWSSubnetID{"subnet-12345", "subnet-67890"},
+					Names: []operatorv1.AWSSubnetName{"name-67890", "name-12345", "name-54321"},
+				},
+			),
+			service:           &corev1.Service{},
+			awsSubnetsEnabled: true,
+			platformStatus:    awsPlatformStatus,
+			expectStatus:      operatorv1.ConditionTrue,
+		},
+		{
+			name: "NLB LoadBalancerService, AWS EIPAllocations nil spec and nil status",
+			ic: loadBalancerIngressControllerWithAWSEIPAllocations(
+				operatorv1.AWSNetworkLoadBalancer,
+				nil,
+				nil,
+			),
+			service:                  &corev1.Service{},
+			awsSubnetsEnabled:        true,
```

Comment:
For all of your unit tests, there's no need to turn on `awsSubnetsEnabled`.

```suggestion
```

Reply:
done